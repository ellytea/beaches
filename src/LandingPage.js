import React, { Component } from 'react';
import './App.css';


class LandingPage extends Component {
  constructor() {
    super();
    this.state = {
      landingPage: true
    };
  }
  
  hide = (event) => {
    this.props.renderBeaches(event);
    this.setState({
      landingPage: false
    });
  }

  render() {
    if (this.state.landingPage) {
      return (
        <div className='landing-page'>
          <div class="sun">
            <div class="ray_box">
              <div class="ray ray1"></div>
              <div class="ray ray2"></div>
              <div class="ray ray3"></div>
              <div class="ray ray4"></div>
              <div class="ray ray5"></div>
              <div class="ray ray6"></div>
              <div class="ray ray7"></div>
              <div class="ray ray8"></div>
              <div class="ray ray9"></div>
              <div class="ray ray10"></div>
            </div>
          </div>
          <h1 className='headline'>BEACHY KEEN</h1>
          <p className='subhead'>SO CAL BEACHES</p>
            <button className='find-button' 
              onClick={this.hide}>Find Your Happy Place
            </button>
        </div>
      );
    } else {
      return (
        <div className='none'></div>
      );
    }
  } 
}

export default LandingPage;