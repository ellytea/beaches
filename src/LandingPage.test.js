import React from 'react'
import LandingPage from './LandingPage'
import { shallow } from 'enzyme'

const renderBeachesMock = jest.fn();

describe ('LandingPage', () => {
  let wrapper = shallow(
    <LandingPage renderBeaches={renderBeachesMock} />
    )

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should invoke renderBeaches and change landingPage state to false when clicked', () => {
        wrapper.find('.landing-button').simulate('click');
        expect(renderBeachesMock).toBeCalled();
        expect(wrapper.state('landingPage')).toEqual(false);
  })

})