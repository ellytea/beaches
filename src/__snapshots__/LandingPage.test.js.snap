// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should match the snapshot with all data passed in 1`] = `
<div
  className="landing-page"
>
  <div
    className="landing-details-section"
  >
    <h1
      className="landing-headline"
    >
      BEACHY KEEN
    </h1>
    <p
      className="landing-subhead"
    >
      SOUTHERN CALIFORNIA BEACHES
    </p>
    <button
      className="landing-button"
      onClick={[Function]}
    >
      Find Your Happy Place
    </button>
  </div>
  <div
    className="sun"
  />
</div>
`;
